src/HasChildren.php

```diff
@@ -94,33 +106,33 @@ protected function getChildModel(array $attributes)
             $attributes[$this->getInheritanceColumn()]
         );
 
-        return new $className((array)$attributes);
+        return new $className((array) $attributes);
     }
 
     public function classFromAlias($aliasOrClass)
     {
-        if (property_exists($this, 'childTypes')) {
-            if (isset($this->childTypes[$aliasOrClass])) {
-                return $this->childTypes[$aliasOrClass];
-            }
+        $types = $this->getChildTypes();
+        if (isset($types[$aliasOrClass])) {
+            return $types[$aliasOrClass];
         }
 
         return $aliasOrClass;
     }
 
     public function classToAlias($className)
     {
-        if (property_exists($this, 'childTypes')) {
-            if (in_array($className, $this->childTypes)) {
-                return array_search($className, $this->childTypes);
-            }
+        if (in_array($className, $this->getChildTypes())) {
+            return array_search($className, $this->getChildTypes());
         }
 
         return $className;
     }
 
     public function getChildTypes()
     {
-        return property_exists($this, 'childTypes') ? $this->childTypes : [];
+        return array_flip(array_merge(
+            config('parental.discovered_children.'.self::class, []),
+            array_flip(property_exists($this, 'childTypes') ? $this->childTypes : [])
```

Comment:
I like the general approach to this file, but without the config-based discovery stuff. Is there any way we can discover the children automatically without costing too much performance wise?

Reply:
Well, I added the DiscoverChildren command since I couldn't find a way to easily know of the existence of the children without explicitely adding them to the childTypes (which is currently pointless unless you use aliases).